src/components/NewWorkspaceModal.js

```diff
@@ -109,82 +98,118 @@ export default ajaxCaller(class NewWorkspaceModal extends Component {
+    const { ajax: { Billing, Groups } } = this.props
+    const [billingProjects, allGroups] = await Promise.all([
+      Billing.listProjects(),
+      Groups.list()
+    ])
+    const usableProjects = _.filter({ creationStatus: 'Ready' }, billingProjects)
+    this.setState(({ namespace }) => ({
+      billingProjects: usableProjects, allGroups,
+      namespace: _.some({ projectName: namespace }, usableProjects) ? namespace : undefined
+    }))
+  })
+
   render() {
-    const { onDismiss, cloneWorkspace } = this.props
+    const { onDismiss, cloneWorkspace, authState: { profile } } = this.props
+    const { trialState } = profile
     const { namespace, name, billingProjects, allGroups, groups, description, nameModified, busy, createError } = this.state
     const existingGroups = this.getRequiredGroups()
+    const hasBillingProjects = !!billingProjects && !!billingProjects.length
+    const hasFreeCredits = trialState === 'Enabled'
     const errors = validate({ namespace, name }, constraints, {
       prettify: v => ({ namespace: 'Billing project', name: 'Name' }[v] || validate.prettify(v))
     })
     return h(Modal, {
-      title: cloneWorkspace ? 'Clone a Workspace' : 'Create a New Workspace',
+      title: busy ? '' : !hasBillingProjects ? 'Set up Billing' : cloneWorkspace ? 'Clone a Workspace' : 'Create a New Workspace',
       onDismiss,
-      okButton: buttonPrimary({
+      showCancel: hasBillingProjects,
+      okButton: busy ? 'Loading...' : hasBillingProjects ? buttonPrimary({
         disabled: errors,
         tooltip: Utils.summarizeErrors(errors),
         onClick: () => this.create()
       }, cloneWorkspace ? 'Clone Workspace' : 'Create Workspace')
+        : hasFreeCredits ?
+          buttonPrimary({
+            onClick: () => {
+              onDismiss()
+              freeCreditsActive.set(true)
+            }
+          }, 'Get Free Credits')
+          : buttonPrimary({
+            onClick: () => Nav.goToPath('billing')
+          }, 'Go to Billing')
     }, [
-      h(RequiredFormLabel, ['Workspace name']),
-      validatedInput({
-        inputProps: {
-          autoFocus: true,
-          placeholder: 'Enter a name',
-          value: name,
-          onChange: e => this.setState({ name: e.target.value, nameModified: true })
-        },
-        error: Utils.summarizeErrors(nameModified && errors && errors.name)
-      }),
-      h(RequiredFormLabel, ['Billing project']),
-      billingProjects && !billingProjects.length ? h(Fragment, [
-        div({ style: { color: colors.red[0] } }, [
-          icon('error', { size: 16 }),
-          ' You must have a billing project associated with your account to create a new workspace.'
-        ]),
-        div({ style: { marginTop: '1rem' } }, [
-          'Billing projects are currently managed through FireCloud. ',
-          link({ target: '_blank', href: billingDoc }, [
-            'Learn how to create a billing project. '
+      !busy && h(Fragment,
```

Comment:
Added this flag so that the contents of the modal don't show up until the modal is ready to display

Reply:
Since we're swapping out the entire contents of the modal, including the buttons and title, it might be cleaner to render them as separate `Modal`s, with a single top-level conditional that chooses the appropriate one (after loading the data).

Reply:
So have 3 or 4 conditional modals? 
1. user has billing project, is creating new workspace
2. user has billing project, is cloning workspace
3. user doesn't have billing project, has free credits
4. user doesn't have billing project, no free credits